Mathlib/Topology/ContinuousMap/Algebra.lean

```diff
@@ -527,18 +527,25 @@ instance instSMul [SMul R M] [ContinuousConstSMul R M] : SMul R C(α, M) :=
   ⟨fun r f => ⟨r • ⇑f, f.continuous.const_smul r⟩⟩
 
 @[to_additive]
-instance [LocallyCompactSpace α] [SMul R M] [ContinuousConstSMul R M] :
-    ContinuousConstSMul R C(α, M) :=
-  ⟨fun γ => continuous_of_continuous_uncurry _ (continuous_eval.const_smul γ)⟩
+instance [SMul R M] [ContinuousConstSMul R M] : ContinuousConstSMul R C(α, M) where
+  continuous_const_smul r := continuous_postcomp ⟨_, continuous_const_smul r⟩
 
 @[to_additive]
-instance [LocallyCompactSpace α] [TopologicalSpace R] [SMul R M] [ContinuousSMul R M] :
-    ContinuousSMul R C(α, M) :=
-  ⟨by
-    refine continuous_of_continuous_uncurry _ ?_
-    have h : Continuous fun x : (R × C(α, M)) × α => x.fst.snd x.snd :=
-      continuous_eval.comp (continuous_snd.prodMap continuous_id)
-    exact (continuous_fst.comp continuous_fst).smul h⟩
+instance [TopologicalSpace R] [SMul R M] [ContinuousSMul R M] :
+    ContinuousSMul R C(α, M) := by
+  constructor
+  simp_rw [continuous_iff_continuousAt, ContinuousAt, ContinuousMap.tendsto_nhds_compactOpen]
+  rintro ⟨r, f⟩ K hK U hU H
+  have : Set.MapsTo (fun p : R × M ↦ p.1 • p.2) ({r} ×ˢ (f '' K)) U := by
+    simpa [Set.MapsTo, forall_comm (α := M), forall_comm (β := _ = _)]
+  have := continuous_smul.tendsto_nhdsSet this hU.mem_nhdsSet_self
+  rw [isCompact_singleton.nhdsSet_prod_eq (hK.image f.continuous), nhdsSet_singleton,
+    Filter.mem_map, ((nhds_basis_opens _).prod (hasBasis_nhdsSet _)).mem_iff] at this
+  obtain ⟨⟨V, W⟩, ⟨⟨hrV, hV⟩, hW, hKW⟩, hVW⟩ := this
+  refine Filter.eventually_of_mem (prod_mem_nhds (hV.mem_nhds hrV)
```

Comment:
(You could use dot notation here. Do you think this would be more readable?)

Reply:
I'm not sure how. Can you explain further?

Reply:
Never mind, I misremembered the hypotheses of Filter.eventually_of_mem.